content/guides/ci-cd/cnbp-knative-argocd.md

```diff
@@ -0,0 +1,263 @@
+
+* You have an account in [Github](https://github.com/) to clone the app *Petclinic*
+
+
+### 1. Prepare a Kubernetes Cluster and clone Sample Application
+
+We will deploy a Kind cluster using Docker Desktop and install [Contour](https://projectcontour.io/) on it to help provide Ingress management. Contour along with [Envoy](https://www.envoyproxy.io/) Proxy will help create service and URL management for Knative. 
+
+```
+brew install kind
+```
+Create a Kubernetes Cluster called *tdp-guide* and set **Kubectl** context to *tdp-guide* 
+
+```
+kind create cluster --name tdp-guide
+kubectl cluster-info --context kind-tdp-guide
+```
+Log onto Github and Fork the repository for our sample app [Petclinic](https://github.com/Boskey/spring-petclinic
+)
+### 2. Install Knative Serving
+
+```
+kubectl apply -f https://github.com/knative/serving/releases/download/v0.22.0/serving-crds.yaml
+kubectl apply -f https://github.com/knative/serving/releases/download/v0.22.0/serving-core.yaml
+```
+### 3. Install Contour Ingress Controller
+```
+kubectl apply -f https://github.com/knative/net-contour/releases/download/v0.22.0/contour.yaml
+kubectl apply -f https://github.com/knative/net-contour/releases/download/v0.22.0/net-contour.yaml
+```
+Change Knative-serving config to use Contour Ingress
+```
+kubectl patch configmap/config-network \
+  --namespace knative-serving \
+  --type merge \
+  --patch '{"data":{"ingress.class":"contour.ingress.networking.knative.dev"}}'
+  ```
+### 4. Install and Configure Cloud Native Buildpack using *Kpack*
+```
+kubectl apply -f https://github.com/pivotal/kpack/releases/download/v0.2.2/release-0.2.2.yaml
+```
+
+Cloud Native Buildpacks will need a Repository to Store container images that it will be building. This could be any OCI compliant repository, for this guide we will use Docker Hub. You can easily create and account in Docker Hub if you don't have one.
+
+We need to create a Docker Hub account credentials *secret* in Kubernetes. Use the below command and change the `docker-username` to the your repo name in Docker Hub. Change the `docker-password` to your account password. 
+
+```
+kubectl create secret docker-registry tutorial-registry-credentials \
+    --docker-username=abc \
+    --docker-password=********* \
+    --docker-server=https://index.docker.io/v1/\
+    --namespace default
+```
+
+Cloud Native Buildpacks create Container images using a `builder` that uses a predefined `stack` of container image layers. You can define custom stack, store and builders. For this guide, we are using standard definitions.
+
+```
+kubectl apply -f https://raw.githubusercontent.com/Boskey/spring-petclinic/main/kpack-config/sa.yaml
+kubectl apply -f https://raw.githubusercontent.com/Boskey/spring-petclinic/main/kpack-config/store.yaml
+kubectl apply -f https://github.com/Boskey/spring-petclinic/blob/main/kpack-config/stack.yaml
+https://raw.githubusercontent.com/Boskey/spring-petclinic/main/kpack-config/builder.yaml
```

Comment:
```suggestion
kubectl apply -f https://raw.githubusercontent.com/Boskey/spring-petclinic/main/kpack-config/builder.yaml
```

I believe this is missing on this line, but feel free to ignore if that's not correct